### PR TITLE
feat(chrome)!: detect Footer automatically with Context

### PR DESCRIPTION
**Pull request by geotrev**
## Description

Removes `hasFooter` prop from `react-chrome`'s `Body` component. Uses React context similar to how other components detect when a child is present in the JSX tree.

## Checklist

- ~~:ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)~~
- [x] :globe_with_meridians: demo is up-to-date (`npm start`)
- ~~:arrow_left: renders as expected with reversed (RTL) direction~~
- ~~:metal: renders as expected with Bedrock CSS (`?bedrock`)~~
- [x] :guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)
- ~~:wheelchair: tested for WCAG 2.1 AA accessibility compliance~~
- ~~:memo: tested in Chrome, Firefox, Safari, and Edge~~

### CODE REVIEW

File: docs/migration.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ consider additional positioning prop support on a case-by-case basis.
 #### @zendeskgarden/react-chrome
 
 - Removed `PRODUCT` type export. Use `IHeaderItemProps['product']` instead.
+- Removed `hasFooter` prop for `Body` (no replacement needed)
```
Comment (jzempel) [inline docs/migration.md:50]: Should we call out that `IBodyProps` is no longer exported?

File: packages/chrome/src/elements/body/Body.tsx [hunk 2]
```diff
@@ -5,27 +5,27 @@
...
 import { BodyContext } from '../../utils/useBodyContext';
 
 /**
  * @extends HTMLAttributes<HTMLDivElement>
  */
-export const Body = React.forwardRef<HTMLDivElement, IBodyProps>(({ hasFooter, ...props }, ref) => {
-  const bodyContextValue = useMemo(() => ({ hasFooter: !!hasFooter }), [hasFooter]);
+export const Body = React.forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
+  (props, ref) => {
+    const [hasFooter, setHasFooter] = useState(true);
```
Comment (jzempel) [inline packages/chrome/src/elements/body/Body.tsx:17]: Shouldn't this initialize to `useState(false)`?

Comment (geotrev) [reply packages/chrome/src/elements/body/Body.tsx:17]: I'm retaining the previous behavior which is [true by default](https://github.com/zendeskgarden/react-components/blob/next/packages/chrome/src/utils/useBodyContext.ts#L15). I'm OK shifting to `false` if that makes more sense though. EIther way the initial state has a 50/50 chance to not match the instructed UI. 🤔

File: docs/migration.md (hunk 1 above)
Comment (geotrev) [reply docs/migration.md:50]: Yep, good call.

**Review by jzempel — APPROVED** [review]

File: packages/chrome/src/elements/body/Body.tsx (hunk 2 above)
Comment (jzempel) [reply packages/chrome/src/elements/body/Body.tsx:17]: Weird. I think the previous was wrong. The `<Body>` definitely doesn't have a footer .. until it does 😅. There are plenty of areas that are `<Footer>`-less. Let's make sure the layout doesn't clip/overflow for any of these cases.

Comment (geotrev) [reply packages/chrome/src/elements/body/Body.tsx]: I agree. Makes more sense to retain a "no footer" base case.

**Pull request merged**